Assets/Shopify/UIToolkit/Components/WebImageCache.cs

```diff
@@ -0,0 +1,163 @@
+ namespace Shopify.UIToolkit {
+    using UnityEngine;
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    public struct CachedWebResource<T> {
+        public readonly string LastModifiedTimestamp;
+        public readonly T Data;
+
+        public CachedWebResource(string lastModifiedTimestamp, T data) {
+            LastModifiedTimestamp = lastModifiedTimestamp;
+            Data = data;
+        }
+    }
+
+    /// <summary>
+    /// This class implements a LRU cache for images (PNG/JPEG) that were downloaded from the web.
+    /// </summary>
+    public class WebImageCache {
+        /// <summary>
+        /// The default estimated memory size limit in bytes.
+        /// </summary>
+        public const int DEFAULT_MEMORY_SIZE_LIMIT = 20971520; // 20 MB in bytes
```

Comment:
Where'd you get this limit from?

Reply:
I pulled it out of thing air 👀 I'll do some analysis over what a good default size would be.

If we keep in the Last-Modified stuff we can also add a disk caching layer to keep more textures available.